ambry-router/src/main/java/com/github/ambry/router/TtlUpdateOperation.java

```diff
@@ -361,8 +361,14 @@ private void checkAndMaybeComplete() {
         operationException.set(new RouterException("TtlUpdateOperation failed possibly because of offline replicas",
             RouterErrorCode.AmbryUnavailable));
       } else if (operationTracker.hasFailedOnNotFound()) {
-        operationException.set(
-            new RouterException("TtlUpdateOperation failed because of BlobNotFound", RouterErrorCode.BlobDoesNotExist));
+        if (operationTracker.getSuccessCount() > 0) {
+          operationException.set(
```

Comment:
let's add a metric here so we would know this happened.

Reply:
Done in the latest commit.